startup_timeout: 10.0

# Message queue item limit
message_queue_item_limit: 3

# Settings for the grid position estimation portion of the GridLineEstimator
grid_estimator:
    # Step size for initial orientation sweep
    theta_step: 0.004

    # Step size for initial translation sweep
    grid_step: 0.01

    # Distance between the center of one gridline and the center of the next
    grid_spacing: 0.229

    # Thickness of each gridline
    grid_line_thickness: 0.038

    # (x-location of gridline intersection) - (x-location of origin)
    grid_zero_offset_x: 0.5

    # (y-location of gridline intersection) - (y-location of origin)
    grid_zero_offset_y: 0.5

    # Number of times to iterate in get1dGridShift
    grid_translation_mean_iterations: 1

    # Threshold to reject lines that aren't aligned with the grid
    line_rejection_angle_threshold: 0.262 # pi/12

    # Minimum altitude to run the detector
    min_extraction_altitude: 0.1

    # Allowed lag between available position and frame timestamp
    allowed_position_stamp_error: 0.1

grid_line_estimator:
    # Should we forgo line counting and just detect lines on the ground?
    debug_line_detector: false

    # Should we spit out a marker vector for the quad's estimated orientation?
    debug_direction: false

    # Should we spit out an edge image for each frame on the `edges` topic?
    debug_edges: true

    # Should we spit out a copy of the input with lines drawn on top on the
    # `lines` topic?
    debug_lines: true

    # Should we spit out markers for the transformed lines?
    debug_line_markers: true

    # Uncomment this to override the height from robot_localization
    # debug_height: 0.22
